### PR TITLE
Add Edit-friendly-name dialog for in-place YAML rewrite

### LINKED ISSUE #2868: Allow friendly_name to be managed in the dashboard
**Describe the problem you have/What new integration you would like**

While working to create https://github.com/esphome/issues/issues/6216, it was noted that from the dashboard there is not a good way to manage the friendly name. The dashboard only exposes the friendly name through the configuration YAML  and this may not be desirable when dealing with factory images. 

Understandably, changing the friendly name in an end device is achievable when the factory image uses the `dashboard_import.package_import_url` attribute, but I am not convinced that there is adequate documentation around  this feature yet (will be requested in another feature request). 

**Please describe your use case for this integration and alternatives you've tried:**

From the end user of a factory image perspective, the end user should be allowed to change the friendly name of the end device without touching the YAML and possibly causing the image to be recompiled. The user could update the friendly name through the 3 dot menu of the device tile. Similar to how the rename hostname occurs today. 

**Additional context**

### PR DESCRIPTION
**Pull request by bdraco**
## Summary

Adds an Edit-friendly-name dialog wired to the new `devices/edit_friendly_name` backend command. Closes esphome/feature-requests#2868. Companion to backend PR esphome/device-builder#390.

## What does this implement/fix?

**The user this is for**: someone gets a Made-For-ESPHome device, adopts it through the dashboard, and wants to call it "Reading Lamp" instead of the manufacturer's default friendly_name. They don't know YAML yet. The goal is two clicks — row menu → dialog → optional install — and the device gets the new label.

The dialog has one input plus an "Install immediately" toggle (default on). On confirm:

1. **Calls `devices/edit_friendly_name`** which rewrites the YAML directly. Substitution-aware (`friendly_name: ${friendly_name}` redirects through the `substitutions:` block, keeping any sensor named `${friendly_name} Power` re-targeted atomically) and safe on YAML-special characters (`Bedroom #2` round-trips through proper quoting). All the YAML-rewrite edge cases covered in #384 (clone) apply here too — we reuse `rewrite_name_or_substitution`.
2. **If `rewritten=false`**: backend signals an idempotent no-op (the leaf already had this value). Surfaces a quiet "unchanged" toast and skips the install. The toast text is i18n'd as `action_friendly_name_unchanged`.
3. **If install toggle is on**: routes through the existing `_openInstallMethod` picker rather than calling `installDevice` directly. The picker already knows how to disable the OTA option for offline / no-OTA devices and offers web-serial / USB / web-download as fallbacks — so users with a Made-For-ESPHome device that can't OTA (or with the device temporarily offline) get clear next-step options instead of a wall of compile errors.

The install-method picker handles every install path the dashboard already knows about, so reusing it avoids a parallel install flow that would have to learn the same edge cases.

## Translations

en / fr / nl ship in this PR.

## Tests

Two new API helper tests pin the wire shape:
- `devices/edit_friendly_name` sent with snake_case args (`new_friendly_name`).
- `rewritten: false` propagated through unchanged so the dashboard handler can branch on it for idempotent edits.

769/769 frontend tests pass; `npm run lint` clean.

**Related issue or feature (if applicable):**

- fixes esphome/feature-requests#2868
- companion backend PR: esphome/device-builder#390

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue) — `bugfix`
- [x] New feature (non-breaking change which adds functionality) — `new-feature`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) — `breaking-change`
- [ ] Refactor (no behaviour change) — `refactor`
- [ ] Documentation only — `docs`
- [ ] Maintenance / chore — `maintenance`
- [ ] CI / workflow change — `ci`
- [ ] Dependencies bump — `dependencies`

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] `npm run lint` passes.
  - [x] `npm run test` passes (769/769).
  - [x] Tests have been added to verify that the new code works (where applicable).

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -126,6 +127,17 @@
...
+    "action_friendly_name_label": "Friendly name",
+    "action_friendly_name_helper": "Shown in Home Assistant and the dashboard list. The hostname stays the same.",
+    "action_friendly_name_required": "Friendly name is required",
+    "action_friendly_name_install_after": "Install immediately so the device announces the new name",
+    "action_friendly_name_install_skipped": "The new name lands in the YAML now; flash the device to apply.",
+    "action_friendly_name_confirm": "Save",
+    "action_friendly_name_success": "Renamed to \"{name}\"",
+    "action_friendly_name_unchanged": "Friendly name unchanged",
+    "action_friendly_name_failed": "Failed to rename \"{name}\": {reason}",
+    "action_friendly_name_install_failed": "Renamed \"{name}\" but install failed: {reason}",
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/translations/en.json]: Fixed in 86d0b93 — removed action_friendly_name_install_failed from en.json. Was a leftover from an earlier draft where install failure had its own toast; routing through _openInstallMethod gave the picker its own error UX so the dedicated key wasn't referenced anywhere.

File: src/translations/fr.json [hunk 2]
```diff
@@ -82,6 +83,17 @@
...
+    "action_friendly_name_label": "Nom convivial",
+    "action_friendly_name_helper": "Affiché dans Home Assistant et la liste du tableau de bord. Le nom d'hôte reste inchangé.",
+    "action_friendly_name_required": "Le nom convivial est requis",
+    "action_friendly_name_install_after": "Installer immédiatement pour que l'appareil annonce le nouveau nom",
+    "action_friendly_name_install_skipped": "Le nouveau nom est enregistré dans le YAML maintenant ; flashez l'appareil pour l'appliquer.",
+    "action_friendly_name_confirm": "Enregistrer",
+    "action_friendly_name_success": "Renommé en \"{name}\"",
+    "action_friendly_name_unchanged": "Nom convivial inchangé",
+    "action_friendly_name_failed": "Échec du renommage de \"{name}\" : {reason}",
+    "action_friendly_name_install_failed": "\"{name}\" renommé mais l'installation a échoué : {reason}",
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/translations/fr.json]: Fixed in 86d0b93 — removed from fr.json too.

File: src/translations/nl.json [hunk 3]
```diff
@@ -82,6 +83,17 @@
...
+    "action_friendly_name_label": "Vriendelijke naam",
+    "action_friendly_name_helper": "Wordt getoond in Home Assistant en de dashboardlijst. De hostnaam blijft hetzelfde.",
+    "action_friendly_name_required": "Vriendelijke naam is verplicht",
+    "action_friendly_name_install_after": "Direct installeren zodat het apparaat de nieuwe naam aankondigt",
+    "action_friendly_name_install_skipped": "De nieuwe naam staat nu in de YAML; flash het apparaat om hem toe te passen.",
+    "action_friendly_name_confirm": "Opslaan",
+    "action_friendly_name_success": "Hernoemd naar \"{name}\"",
+    "action_friendly_name_unchanged": "Vriendelijke naam ongewijzigd",
+    "action_friendly_name_failed": "Hernoemen van \"{name}\" mislukt: {reason}",
+    "action_friendly_name_install_failed": "\"{name}\" hernoemd maar installatie mislukt: {reason}",
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/translations/nl.json]: Fixed in 86d0b93 — removed from nl.json too.

File: src/pages/dashboard.ts [hunk 4]
```diff
@@ -286,6 +288,27 @@ export class ESPHomePageDashboard extends LitElement {
...
+       a rename, stale ``state`` after a flap, stale ``ip`` after
+       a DHCP renew). Lookup is by ``configuration`` since that's
+       the stable identity the WS reducer keys on too. */
+    if (changed.has("_devices") && this._drawerDevice) {
+      const live = this._devices.find(
+        (d) => d.configuration === this._drawerDevice!.configuration,
+      );
+      if (live && live !== this._drawerDevice) {
+        this._drawerDevice = live;
+      } else if (!live) {
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/pages/dashboard.ts:306]: Real catch — fixed in dd81fdb. Updated drawer-content's updated() to compare device.configuration (the stable WS identity) rather than reference identity. Same-device WS pushes still flow through to a re-render so the displayed friendly_name / IP / version etc. update, but per-device state (_ipExpanded) stays stable and the reachability subscription doesn't rotate. Only an actual configuration change (drawer reused for a different device, or null→device transition) triggers the resets / resubscribe.
